src/inmanta/module.py

```diff
@@ -1657,7 +1664,14 @@ def get_relation_precedence_rules(self) -> List[RelationPrecedenceRule]:
         return [RelationPrecedenceRule.from_string(rule_as_str) for rule_as_str in self.relation_precedence_policy]
 
     def get_index_urls(self) -> List[str]:
-        return [repo.url for repo in self.repo if repo.type == ModuleRepoType.package]
+        # Once setting repos with type package is no longer supported, this method can return self.pip.index_url alone.
+        index_urls_deprecated_option: List[str] = [repo.url for repo in self.repo if repo.type == ModuleRepoType.package]
+        if all([index_urls_deprecated_option, self.pip.index_url]):
+            LOGGER.warning(
+                "Pip indexes are configured in two places. Setting them through the `repo -> url` option has been "
+                "deprecated in favour of the `pip -> index_url` option."
+            )
+        return self.pip.index_url + index_urls_deprecated_option
```

Comment:
Don't we need to make sure that we don't have any duplicates in this list.

Reply:
Added a cast to `set` and then back to `list`

Reply:
I would not do this because it affects which is the `--index-url` and which are `--extra-index-url`. Afaik pip doesn't really treat them differently but I wouldn't count on that. Going through `dict` instead could work because it keeps insertion order. I don't know off the top of my head how it treats duplicate keys and how it affects insertion order so that should be checked.

Reply:
Ok, I'll make it so order is preserved and document clearly the behaviour regarding `--index-url` vs `--extra-index-url`

Reply:
I don't think we need to document it. That might strengthen the impression that there is a difference. I think if we make sure to document that we strongly discourage using more than one index in the first place we're good. And I think we already have that, or we have a ticket for it.

Reply:
Checked just now, in case of duplicate keys, the value is indeed updated but it has no impact on the recorded insertion order:
```python
import itertools

l1 = ["a1", "b2", "c3"]
l2 = ["c3", "d4", "a1"]

a = list({value: None for value in itertools.chain(l1, l2)})

print(a) #  ['a1', 'b2', 'c3', 'd4']

out = {}
for i, value in enumerate(itertools.chain(l1, l2)):
    out[value] = i

print(out) #  {'a1': 5, 'b2': 1, 'c3': 3, 'd4': 4}
print(list(out)) #  ['a1', 'b2', 'c3', 'd4']


```